pkg/authorization/authorizer/attributes.go

```diff
@@ -34,6 +34,12 @@ func (a DefaultAuthorizationAttributes) RuleMatches(rule authorizationapi.Policy
 
 		if a.resourceMatches(allowedResourceTypes) {
 			if a.nameMatches(rule.ResourceNames) {
+				// this rule matches the request, so we should check the additional restrictions to be sure that it's allowed
+				switch rule.AttributeRestrictions.Object.(type) {
```

Comment:
if we have attribute restrictions, and they aren't a recognized type, we should return false